File: 2016/gara1/ricostruzioni/att/ricostruzioni.c
#include <stdio.h>
#include <assert.h>

#define MAXN 100000

int spiana(int N, int K, int A[]) {
    // Mettete qui il codice della soluzione
    return 42;
}


int A[MAXN];

int main() {
    FILE *fr, *fw;
    int N, K, i;

    fr = fopen("input.txt", "r");
    fw = fopen("output.txt", "w");
    assert(2 == fscanf(fr, "%d %d", &N, &K));
    for(i=0; i<N; i++)
        assert(1 == fscanf(fr, "%d", &A[i]));

    fprintf(fw, "%d\n", spiana(N, K, A));
    fclose(fr);
    fclose(fw);
    return 0;
}

File: ed2/gara1/ricostruzioni/att/ricostruzioni.pas
function spiana(N, K: longint; var A: array of longint): longint;
begin
    (* Mettete qui il codice della soluzione *)
    spiana := 42;
end;


const
    MAXN = 100000;
var
    N, K, i : longint;
    A       : array[0..MAXN-1] of longint;
    fr, fw  : text;

begin
    assign(fr, 'input.txt');
    assign(fw, 'output.txt');
    reset(fr);
    rewrite(fw);
    readln(fr, N, K);
    for i:=0 to N-1 do
        read(fr, A[i]);

    writeln(fw, spiana(N, K, A));
    close(fr);
    close(fw);
end.
